### PR TITLE
Fix Integrity Constraint Violation exception on delete command

### LINKED ISSUE #101: Deleting Oauth Clients throws Integrity Constraint Violation exception
When deleting a Client from the command line an exception is thrown:

`php bin/console trikoder:oauth2:delete-client test`

```
In AbstractMySQLDriver.php line 49:
                                                                                                                                                                                                                        
  An exception occurred while executing 'DELETE FROM oauth2_client WHERE identifier = ?' with params ["test"]:                                                                                                     
                                                                                                                                                                                                                        
  SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`db`.`oauth2_access_token`, CONSTRAINT `FK_454D9673C7440455` FOREIGN KEY (`c  
  lient`) REFERENCES `oauth2_client` (`identifier`))                                                                                                                                                                    
                                                                                                                                                                                                                        
In PDOStatement.php line 119:
                                                                                                                                                                                                                        
  SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`db`.`oauth2_access_token`, CONSTRAINT `FK_454D9673C7440455` FOREIGN KEY (`c  
  lient`) REFERENCES `oauth2_client` (`identifier`))                                                                                                                                                                    
                                                                                                                                                                                                                        
In PDOStatement.php line 117:
                                                                                                                                                                                                                        
  SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails (`db`.`oauth2_access_token`, CONSTRAINT `FK_454D9673C7440455` FOREIGN KEY (`c  
  lient`) REFERENCES `oauth2_client` (`identifier`))                                                                                                                                                                    
                                                                                                                                                                                                                        
```
This makes sense, because there are still records in the `oauth2_access_token` database. It could be solved 2 ways, by setting cascade DELETE on the entity, or by manually deleting all Access Token records with the related identifier from the database.

### PR DESCRIPTION
**Pull request by Allypost**
Fixes #101

### CODE REVIEW

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+namespace Trikoder\Bundle\OAuth2Bundle\Tests\Acceptance;
+
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Trikoder\Bundle\OAuth2Bundle\Manager\Doctrine\ClientManager as DoctrineClientManager;
+use Trikoder\Bundle\OAuth2Bundle\Model\AccessToken;
+use Trikoder\Bundle\OAuth2Bundle\Model\Client;
+use Trikoder\Bundle\OAuth2Bundle\Model\RefreshToken;
+
+class DoctrineClientManagerTest extends AbstractAcceptanceTest
```
Comment (HypeMC) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: @Allypost Why not add `@covers` here as well ? Also, `final`.

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        );
+
+        // The entity manager has to be cleared manually
+        // because it doesn't process deep integrity constraints
+        $em->clear();
+
+        /** @var $refreshToken RefreshToken */
+        $refreshToken = $em
+            ->getRepository(RefreshToken::class)
+            ->findOneBy(
```
Comment (HypeMC) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: @Allypost https://www.doctrine-project.org/api/orm/latest/Doctrine/ORM/EntityRepository.html#method_find

File: Tests/Acceptance/DoctrineClientManagerTest.php (hunk 1 above)
Comment (Allypost) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: @HypeMC Fixed.

File: Tests/Acceptance/DoctrineClientManagerTest.php (hunk 2 above)
Comment (Allypost) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: @HypeMC Fixed.

File: Tests/Acceptance/DoctrineClientManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+use Trikoder\Bundle\OAuth2Bundle\Manager\Doctrine\ClientManager as DoctrineClientManager;
+use Trikoder\Bundle\OAuth2Bundle\Model\AccessToken;
+use Trikoder\Bundle\OAuth2Bundle\Model\Client;
+use Trikoder\Bundle\OAuth2Bundle\Model\RefreshToken;
+
+/**
+ * @covers \Trikoder\Bundle\OAuth2Bundle\Manager\Doctrine\ClientManager
```
Comment (HypeMC) [inline Tests/Acceptance/DoctrineClientManagerTest.php]: @Allypost I think the same TODO applies here as in DoctrineAccessTokenManagerTest & DoctrineRefreshTokenManagerTest, could you please add it.

Comment (Allypost) [reply Tests/Acceptance/DoctrineClientManagerTest.php]: @HypeMC Fixed. Also added `@covers` tags to the aforementioned classes.

**Pull request merged**